lib/AST/ASTContext.cpp

```diff
@@ -275,8 +275,16 @@ struct ASTContext::Implementation {
     ~Arena() {
       for (auto &conformance : SpecializedConformances)
         conformance.~SpecializedProtocolConformance();
+      // Work around MSVC warning: local variable is initialized but
+      // not referenced.
+#if defined(_MSC_VER)
+#pragma warning (disable: 4189)
```

Comment:
This seems to be a bug in MSVC: I've been trying hard to reproduce this came, but can't.